sources/iTermColorMap.m

```diff
@@ -86,19 +87,30 @@ - (void)setMutingAmount:(double)mutingAmount {
 }
 
 - (void)setColor:(NSColor *)theColor forKey:(iTermColorMapKey)theKey {
-    if (!theColor || theColor == _map[@(theKey)] || theKey >= kColorMap24bitBase) {
+    if (theKey >= kColorMap24bitBase)
+        return;
+
+    if ( !theColor ) {
+        [_map removeObjectForKey:@(theKey)];
         return;
     }
+
+    if (theColor == _map[@(theKey)])
+        return;
+
     if (theKey == kColorMapBackground) {
         _backgroundRed = [theColor redComponent];
         _backgroundGreen = [theColor greenComponent];
         _backgroundBlue = [theColor blueComponent];
     }
-    theColor = [theColor colorUsingColorSpaceName:NSCalibratedRGBColorSpace];
-    _map[@(theKey)] = theColor;
+
     if (theKey == kColorMapBackground) {
         _backgroundBrightness = [theColor perceivedBrightness];
     }
+
+    theColor = [theColor colorUsingColorSpaceName:NSCalibratedRGBColorSpace];
```

Comment:
Did you move this intentionally? I think backgroundBrightness should be calculated from the RGB colorspace since that is what gets displayed.